yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -372,7 +372,14 @@ private[spark] class ApplicationMaster(
             case i: InterruptedException =>
             case e: Throwable => {
               failureCount += 1
-              if (!NonFatal(e) || failureCount >= reporterMaxFailures) {
+              if ("org.apache.hadoop.yarn.exceptions.ApplicationAttemptNotFoundException".equals(
+                e.getClass().getName())) {
+                val message = "ApplicationAttemptNotFoundException was thrown " +
+                  "from Reporter thread."
```

Comment:
I don't think this message is very useful. You can see the exception from the stack trace anyway...